diff -ur freetype-2.5.2/include/config/ftoption.h freetype-2.5.2-enable-spr/include/config/ftoption.h
--- freetype-2.5.2/include/config/ftoption.h	2013-12-08 19:40:19.000000000 +0100
+++ freetype-2.5.2-enable-spr/include/config/ftoption.h	2014-01-17 19:17:34.000000000 +0100
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
